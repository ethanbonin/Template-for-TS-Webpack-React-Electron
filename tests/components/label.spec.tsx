import React from 'react';
import LabelProperties, {
    HeaderVariants,
} from '@/renderer/types/components/labelProperties';
import Label from '@/renderer/components/label/label';
import { shallow } from 'enzyme';

describe('<Label />', () => {
    it('renders Label', () => {
        const labelProps: LabelProperties = {
            id: 'label-test',
            text: 'Text Label',
            variant: HeaderVariants.h1,
        };

        const wrapper = shallow(<Label label={labelProps} />);
        expect(wrapper).toMatchSnapshot();
    });

    it('renders all different variants', () => {
        for (const value in HeaderVariants) {
            const labelProps: LabelProperties = {
                id: 'label-test',
                text: 'Text Label',
                variant: value,
            };

            const wrapper = shallow(<Label label={labelProps} />);
            expect(wrapper.type()).toEqual(value);
        }
    });

    it('renders text', () => {
        const labelProps: LabelProperties = {
            id: 'label-test',
            text: 'Text Label',
            variant: HeaderVariants.h1,
        };

        const wrapper = shallow(<Label label={labelProps} />);
        expect(wrapper.text()).toEqual(labelProps.text);
    });
});
