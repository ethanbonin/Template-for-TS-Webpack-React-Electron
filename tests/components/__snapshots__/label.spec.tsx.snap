// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> renders Label 1`] = `ShallowWrapper {}`;
